### PR TITLE
Fix for issue #15 (NullPointerException in shutdown hook)

### PR DESCRIPTION
**Pull request by remkop**
Please consider merging these changes into allocation-instrumenter master and creating a new release.

This fixes issue #15 (NullPointerException in shutdown hook).

Many thanks from the Log4j 2 project!

### CODE REVIEW

**Comment by remkop** [comment]
I signed the CLA.

File: src/main/java/com/google/monitoring/runtime/instrumentation/NullInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+/**
+ * Implementation of the {@code Instrumentation} interface that does nothing.
+ * Intended to be used in {@link AllocationRecorder} to work around a bug in certain JVMs
+ * where calling {@code Instrumentation.getObjectSize()} during JVM shutdown triggers a
+ * JVM-crashing assert.
+ */
+class NullInstrumentation implements Instrumentation {
+  @Override
+  public void addTransformer(final ClassFileTransformer transformer, final boolean canRetransform) {
+  }
```
Comment (fischman) [inline src/main/java/com/google/monitoring/runtime/instrumentation/NullInstrumentation.java]: IWBN to throw a RuntimeException if anyone ever called any method of this class other than getObjectSize().

**Comment by fischman** [comment]
FWIW change LGTM modulo minor suggestion.
I'm not an owner/maintainer of this repository, though, so that doesn't necessarily mean much.

File: src/main/java/com/google/monitoring/runtime/instrumentation/NullInstrumentation.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2009 Google Inc.
```
Comment (garydgregory) [inline src/main/java/com/google/monitoring/runtime/instrumentation/NullInstrumentation.java]: Shouldn't the copyright be 2016?

**Comment by jhmanson** [comment]
I commented on the pull request.  I don't like the approach of adding a new
class here - I don't see the need - but I'm open to being wrong about it.

Jeremy

**Comment by jhmanson** [comment]
This seems okay.  Did you test it to make sure that it avoids the NPE?

**Comment by remkop** [comment]
Well, I've never reproduced the issue on my box to be honest. It only happens randomly on the build server. So the honest answer to that is no. But based on the NPE stacktrace this should do the trick.

**Comment by remkop** [comment]
Don't hesitate to let me know if there is anything more I can do.

**Pull request merged**